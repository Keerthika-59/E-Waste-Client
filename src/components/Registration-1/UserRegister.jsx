import React, { useState } from 'react'
import { Form, Button, FormLabel } from 'react-bootstrap';

const UserRegister = () => {

    const [password, setPassword] = useState('')
    const [cpassword, setCPassword] = useState('')

    return (
        <Form className="w-75 mx-auto">
            <h3 className="py-2 px-2"> User Registeration</h3>

            <div className="row">
                <div className="col-md-6 col-sm-6 col-xs-12">
                    <div className="form-group">
                        <FormLabel>Name <em style={{ color: 'red' }}> * </em>  </FormLabel>
                        <input type="text"
                            className="form-control"
                            placeholder="Enter Name"
                            required={true}
                        />
                    </div>

                    <div className="form-group">
                        <FormLabel>Phone Number <em style={{ color: 'red' }}> * </em>  </FormLabel>
                        <input type="text"
                            className="form-control"
                            placeholder="Enter Mobile Number"
                            required={true}
                            maxLength={10}
                        />
                    </div>


                    <div className="form-group">
                        <FormLabel>Email <em style={{ color: 'red' }}> * </em>  </FormLabel>

                        <input type="email"
                            className="form-control"
                            placeholder="Enter email"
                            required={true}
                        />

                    </div>

                    <div className="form-group px">
                        <FormLabel> Gender <em style={{ color: 'red' }}> * </em>  </FormLabel>
                        <Form.Control as="select">
                            <option value='Male'> Male </option>
                            <option value='Female' > Female </option>
                            <option value='others'> Others </option>
                        </Form.Control>
                    </div>
                    
                </div>

                <div className="col-md-6 col-sm-6 col-xs-12">

                    <div className="form-group">
                        <FormLabel>City <em style={{ color: 'red' }}> * </em>  </FormLabel>
                        <input type="text"
                            className="form-control"
                            placeholder="Enter City"
                            required={true}
                        />
                    </div>

                    <div className="form-group">
                        <FormLabel> Address <em style={{ color: 'red' }}> * </em>  </FormLabel>
                        <input type="text"
                            className="form-control"
                            placeholder="Enter Building/Apartment"
                            required={true}
                        />
                    </div>

                    <div className="row">
                        <div className="col">
                            <div className="form-group">
                                <FormLabel> LandMark </FormLabel>
                                <input type="text"
                                    className="form-control"
                                    placeholder="Enter Landmark"
                                />
                            </div>
                        </div>

                        <div className="col">
                            <div className="form-group">
                                <FormLabel> Pin Code <em style={{ color: 'red' }}> * </em>  </FormLabel>
                                <input type="text"
                                    className="form-control"
                                    placeholder="Enter Pin Code"
                                    required={true}
                                />
                            </div>
                        </div>
                    </div>



                    <div className="form-group">
                        <FormLabel>Password <em style={{ color: 'red' }}> * </em> </FormLabel>
                        <input
                            type="password"
                            onChange={(e) => { setPassword(e.target.value) }}
                            className="form-control"
                            placeholder="Enter password"
                            required={true}
                        />
                    </div>
                    
                    <div className="form-group">
                        <FormLabel>Confirm Password <em style={{ color: 'red' }}> * </em> </FormLabel>
                        <input onChange={(e) => { setCPassword(e.target.value) }}
                            type="password"
                            className="form-control"
                            placeholder="Enter confirm password"
                            required={true}
                        />
                        <p>
                            {password.length > 0 && cpassword.length > 0 ? (password === cpassword ? 'Password Matched☑️' : 'Password Not Matched yet') : ''}
                        </p>
                    </div>

                </div>
            </div>

            <Button disabled={false} type="submit"
                className=" my-4 btn-primary btn-lg btn-block"
            > Register
            </Button>
        </Form>
    )
}

export default UserRegister
