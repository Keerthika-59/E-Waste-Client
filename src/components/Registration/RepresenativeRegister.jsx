import React, { useState,useEffect } from 'react'
import { Alert, Modal, Button } from 'react-bootstrap';
import APIHelper from '../API/apihelper2';
import '../Registration/style.css';
import { BrowserRouter, Link } from 'react-router-dom';
import { Formik, Field, Form, ErrorMessage } from 'formik';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import * as Yup from 'yup';
// import toast, { Toaster } from 'react-hot-toast';

const notify = () => toast.success('Representative registration successful!',{position: toast.POSITION.TOP_RIGHT}, {autoClose:5000});
const notify1 = () => toast.error('Error in Representative registration !',{position: toast.POSITION.TOP_RIGHT}, {autoClose:5000});
const phoneRegExp = /^[0-9]{10}$/g;
const nameRegExp = /^[a-zA-Z ]{2,30}$/;

const RepresenativeRegister = () => {
    
    useEffect(() => {
    window.scrollTo(0, 0)
  }, []);


  return (

    <Formik
      initialValues={{
        name: '',
        email: '',
        mobileNumber: '',
        password: '',
        confirmPassword: '',
        city: '',
        address: '',
        gender: ''
      }}
      onSubmit={async (values, { setSubmitting, resetForm }) => {

        const data = {
          name: values.name,
          phoneNumber: values.mobileNumber,
          email: values.email,
          gender: values.gender,
          city: values.city,
          address: values.address,
          password: values.password,
        };
        try {

          await APIHelper.registerUsers(data);
          resetForm({});
          setTimeout(() => {
            //   alert("Form Submitted");
            notify();
            setSubmitting(false);
          }, 1000);
        } catch (err) {
          // alert(err.response.data.errorMessage);
          notify1();
        }

      }}

      validationSchema={Yup.object({
        name: Yup.string()
          .required('Name is required')
          .matches(nameRegExp, 'Name is not Valid'),
        email: Yup.string()
          .email('Invalid email address')
          .required('Email is required'),

        password: Yup.string()
          .required('Password is Required')
          .min(8, 'Password is too short - should be 8 chars minimum.')
          .max(16, 'Password is too long - should be 16 chars maximum.')
          .matches(/(?=.*[0-9])/, "Password must contain a number."),

        confirmPassword: Yup.string()
          .required('Confirm Password is Required')
          .min(8, 'Password is too short - should be 8 chars minimum.')
          .max(16, 'Password is too long - should be 16 chars maximum.')
          .oneOf([Yup.ref('password'), 'Password not matching...']),

        mobileNumber: Yup.string()
          .matches(phoneRegExp, 'Mobile Number is not Valid')
          .min(10, "Too short")
          .max(10, "Too long"),

        gender: Yup.string()
          .required('Gender is required'),

        city: Yup.string()
          .required('City Name is required'),

        address: Yup.string()
          .required('Address is Required')

      })}

    >

      { (formik, values, isSubmitting, resetForm) => (

        <Form>
          <div className="container-fluid px-1 px-md-5 px-lg-1 px-xl-5 py-5 mx-auto">
            <div className="card card0 border-0">
              <div className="row d-flex">
                <div className="col-lg-6">
                  <div className="card1 pb-5">

                    <div className="row px-3 justify-content-center mt-4 mb-5 border-line"> <img src={`${process.env.PUBLIC_URL}/images/picture.jpg`} alt="image" className="image" /> </div>
                  </div>
                </div>

                <div className="col-lg-6">

                  <div className="card2 card border-0 px-4 py-5">

                    <div className="row mb-4 px-3">
                      <h3 className="heading">Representative Sign Up</h3>
                    </div>

                    <div className="row px-3"> <label className="mb-1">
                      <h6 className="mb-0 text-sm">Name</h6>
                    </label>

                      <Field name="name" placeholder="Enter your Name" className={(formik.touched.name && formik.errors.name) ? 'form-control is-invalid' : 'form-control'} type="name" />
                      {formik.touched.name && formik.errors.name ? (
                        <div className="invalid-feedback">{formik.errors.name}</div>
                      ) : null}

                    </div>
                    <br></br>
                    <div className="row px-3">
                      <label className="mb-1">
                        <h6 className="mb-0 text-sm">Phone number</h6>
                      </label>

                      <Field name="mobileNumber" placeholder="Enter your Mobile Number" className={(formik.touched.mobileNumber && formik.errors.mobileNumber) ? 'form-control is-invalid' : 'form-control'} type="text" />
                      {formik.touched.mobileNumber && formik.errors.mobileNumber ? (
                        <div className="invalid-feedback">{formik.errors.mobileNumber}</div>
                      ) : null}

                    </div>
                    <br />
                    <div className="row px-3">

                      <label className="mb-1">
                        <h6 className="mb-0 text-sm">Email Address</h6>
                      </label>

                      <Field name="email" placeholder="Enter your Email" className={(formik.touched.email && formik.errors.email) ? 'form-control is-invalid' : 'form-control'} type="email" />
                      {formik.touched.email && formik.errors.email ? (
                        <div className="invalid-feedback">{formik.errors.email}</div>
                      ) : null}

                    </div>
                    <br />

                    <div>  <h6 className="mb-0 text-sm">Id Proof</h6>
                      <input
                        type="file"
                        accept=".png, .jpg, .jpeg"
                        name="photo"
                      // onChange={handlePhoto}
                      />
                    </div>
                    <br />


                    <div className="row px-3">
                      <label className="mb-1">
                        <h6 className="mb-0 text-sm"> Password</h6>

                      </label>
                      <Field name="password" placeholder="Enter your Password" className={(formik.touched.password && formik.errors.password) ? 'form-control is-invalid' : 'form-control'} type="password" />
                      {formik.touched.password && formik.errors.password ? (
                        <div className="invalid-feedback">{formik.errors.password}</div>
                      ) : null}

                    </div>

                    <br />
                    <div className="row px-3">

                      <label className="mb-1">
                        <h6 className="mb-0 text-sm">Confirm Password</h6>
                      </label>

                      <Field name="confirmPassword" placeholder="Enter your Confirm Password" className={(formik.touched.password && formik.errors.password) ? 'form-control is-invalid' : 'form-control'} type="password" />
                      {formik.touched.password && formik.errors.confirmPassword ? (
                        <div className="invalid-feedback">{formik.errors.confirmPassword}</div>
                      ) : null}

                      {
                        formik.values.password && formik.values.confirmPassword ? (
                          formik.values.password !== formik.values.confirmPassword ?
                            (<p style={{ color: 'red' }}> Password does not match </p>) : (<p style={{ color: 'green' }} > Password Matched  </p>)
                        ) : ''
                      }

                    </div>
                    <br />
                    <p>
                      <div className="row px-3">
                        <label className="mb-1">
                          <h6 className="mb-0 text-sm">Gender</h6>
                        </label>
                        <div role="group" aria-labelledby="my-radio-group">
                          <label className="px-3">
                          <br/>
                            <Field type="radio" name="gender" value="Male" /> Male
                                                        </label>
                          <label className="px-3">
                            <Field type="radio" name="gender" value="Female"/> Female
                          </label>

                          <label className="px-3">
                            <Field type="radio" name="gender" value="Others" /> Others
                                                    </label>
                        </div>
                      </div>
                      <br />

                      <div className="row px-3">
                        <label className="mb-1">
                          <h6 className="mb-0 text-sm">City</h6>
                        </label>

                        <Field name="city" placeholder="Enter your City" className={(formik.touched.city && formik.errors.city) ? 'form-control is-invalid' : 'form-control'} type="text" />
                        {formik.touched.city && formik.errors.city ? (
                          <div className="invalid-feedback">{formik.errors.city}</div>
                        ) : null}

                      </div>
                      <br />

                      <div className="row px-3">
                        <label className="mb-1">
                          <h6 className="mb-0 text-sm">Address</h6>
                        </label>

                        <Field name="address" placeholder="Enter your Address" className={(formik.touched.address && formik.errors.address) ? 'form-control is-invalid' : 'form-control'} type="text" />
                        {formik.touched.address && formik.errors.address ? (
                          <div className="invalid-feedback">{formik.errors.address}</div>
                        ) : null}
                      </div>
                      <br />
                      <div className="row mb-3 px-3"> <button type="submit" className="btn btn-blue text-center">Register</button> </div>
                      <ToastContainer limit={1} />
                      <div className="row mb-4 px-3"> <small className="font-weight-bold">Already have an account?  <Link to="/RepresentativeLogIn"> Login </Link>   </small> </div>

                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>

        </Form>

      )}

    </Formik>
  )
}

export default RepresenativeRegister;
