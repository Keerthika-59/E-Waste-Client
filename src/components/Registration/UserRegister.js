import React, { useState } from 'react'
// import { Form, Button, FormLabel } from 'react-bootstrap';
import '../Registration/style.css';
// const Download=require('./Registration/download.jpg');
const UserRegister = () => {

    const [password, setPassword] = useState('')
    const [cpassword, setCPassword] = useState('')

    return (
        <div className="container-fluid px-1 px-md-5 px-lg-1 px-xl-5 py-5 mx-auto">
        <div className="card card0 border-0">
            <div className="row d-flex">
                <div className="col-lg-6">
                    <div className="card1 pb-5">
                        
                        <div className="row px-3 justify-content-center mt-4 mb-5 border-line"> <img src={`${process.env.PUBLIC_URL}/assets/images/picture.jpg`} alt="image" className="image"/> </div>
                    </div>
                </div>
                <div className="col-lg-6">
                    <div className="card2 card border-0 px-4 py-5">
                        <div className="row mb-4 px-3">
                            <h3 className="heading">User Sign Up</h3>
                        </div>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Name</h6>
                            </label> <input className="mb-4" type="text" name="text" placeholder="Enter a valid name"/> </div>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Phone number</h6>
                            </label> <input className="mb-4" type="text" name="text" placeholder="Enter a valid phone number"/> </div>
                        
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Email Address</h6>
                            </label> <input className="mb-4" type="text" name="email" placeholder="Enter a valid email address"/> </div>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm"> Password</h6>
                            </label> <input type="password" name="password" placeholder="Enter password " onChange={(e) => { setPassword(e.target.value) }}/> </div> <br/>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Confirm Password</h6>
                            </label> <input type="password" name="password" placeholder="Enter password again" onChange={(e) => { setCPassword(e.target.value) }}/> </div>
                        <br/>
                        <p>
                             {password.length > 0 && cpassword.length > 0 ? (password === cpassword ? 'Password Matched☑️' : 'Password Not Matched yet') : ''}                       </p> 
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Gender</h6>
                         </label> 
                         <div class="custom-control custom-radio">
                        <input type="radio" class="custom-control-input" id="defaultUnchecked" name="defaultExampleRadios"/>
                        <label class="custom-control-label" for="defaultUnchecked">Male</label>
                        </div>

                        
                        <div class="custom-control custom-radio">
                        <input type="radio" class="custom-control-input" id="defaultChecked" name="defaultExampleRadios" />
                        <label class="custom-control-label" for="defaultChecked">Female</label>
                        </div>

                        <div class="custom-control custom-radio">
                        <input type="radio" class="custom-control-input" id="defaultChecked" name="defaultExampleRadios" />
                        <label class="custom-control-label" for="defaultChecked">Others</label>
                        </div>
                             {/*<select name="gender">
                                <option>Select your gender</option>
                                <option>Male</option>
                                <option>Female</option>
                                <option>Others</option>
                            </select> */}
                        </div> <br/>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">City</h6>
                            </label> <input className="mb-4" type="text" name="city" placeholder="Enter a city name"/> </div>
                        <div className="row px-3"> <label className="mb-1">
                                <h6 className="mb-0 text-sm">Address</h6>
                            </label> <textarea className="mb-4" type="textarea" name="address" placeholder="Enter a valid address"/> </div>
                            <div className="row mb-3 px-3"> <button type="submit" className="btn btn-blue text-center">Register</button> </div>
                        <div className="row mb-4 px-3"> <small className="font-weight-bold">Already have an account? <a className="text-danger ">Login</a></small> </div>
                 
                        </div>
                        </div>
                </div>
            </div>
       </div>
        )
    }

export default UserRegister;