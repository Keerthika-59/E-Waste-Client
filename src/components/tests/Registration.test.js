import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import UserRegister from '../Registration/UserRegister';
import RepresenativeRegister from '../Registration/RepresenativeRegister';

test('renders test NavBar', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<UserRegister/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });
   test('should test UserRegister component', () => {
    const wrapper = shallow(<UserRegister />);
    expect(wrapper).toMatchSnapshot();
   });

   
test('renders test admin home', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<RepresenativeRegister/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });
   test('should test RepresenativeRegister component', () => {
    const wrapper = shallow(<RepresenativeRegister />);
    expect(wrapper).toMatchSnapshot();
   });