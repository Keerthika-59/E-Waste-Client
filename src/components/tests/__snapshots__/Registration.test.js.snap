// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test RepresenativeRegister component 1`] = `
<Formik
  initialValues={
    Object {
      "address": "",
      "city": "",
      "confirmPassword": "",
      "email": "",
      "gender": "",
      "idProof": "",
      "mobileNumber": "",
      "name": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_excludedEdges": Array [],
      "_mutate": undefined,
      "_nodes": Array [
        "address",
        "city",
        "gender",
        "mobileNumber",
        "confirmPassword",
        "password",
        "email",
        "name",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "address": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "city": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "confirmPassword": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {
              "Password not matching...",
            },
            "refs": Map {
              "password" => Reference {
                "getter": [Function],
                "isContext": false,
                "isSibling": true,
                "isValue": false,
                "key": "password",
                "map": undefined,
                "path": "password",
              },
            },
          },
          "_whitelistError": [Function],
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "email": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "email": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "gender": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "mobileNumber": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "max": true,
            "min": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "optional",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "name": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "password": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;

exports[`should test UserRegister component 1`] = `
<Formik
  initialValues={
    Object {
      "address": "",
      "city": "",
      "confirmPassword": "",
      "email": "",
      "gender": "",
      "mobileNumber": "",
      "name": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_excludedEdges": Array [],
      "_mutate": undefined,
      "_nodes": Array [
        "address",
        "city",
        "gender",
        "mobileNumber",
        "confirmPassword",
        "password",
        "email",
        "name",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "address": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "city": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "confirmPassword": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {
              "Password not matching...",
            },
            "refs": Map {
              "password" => Reference {
                "getter": [Function],
                "isContext": false,
                "isSibling": true,
                "isValue": false,
                "key": "password",
                "map": undefined,
                "path": "password",
              },
            },
          },
          "_whitelistError": [Function],
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "email": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "email": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "gender": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "mobileNumber": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "max": true,
            "min": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "optional",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "name": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "password": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
